skyd/api.go

```diff
@@ -68,44 +103,66 @@ func NewAPI(nginxHost string, nginxPort int, skydHost, skydPassword string, skyd
 	}, nil
 }
 
-// BlockSkylinks will perform an API call to skyd to block the given skylinks
-func (api *api) BlockHashes(hashes []string) error {
-	// Build the call to skyd.
-	reqBody := skyapi.SkynetBlocklistPOST{
-		Add:    hashes,
+// BlockHashes will perform an API call to skyd to block the given hashes. It
+// returns which hashes were blocked, which hashes were invalid and potentially
+// an error.
+func (api *api) BlockHashes(hashes []database.Hash) ([]database.Hash, []database.Hash, error) {
+	api.staticLogger.Debugf("blocking %v hashes", len(hashes))
+
+	// convert the hashes to strings
+	adds := make([]string, len(hashes))
+	for h, hash := range hashes {
+		adds[h] = hash.String()
+	}
+
+	// build the call to skyd.
+	reqBody, err := json.Marshal(skyapi.SkynetBlocklistPOST{
+		Add:    adds,
 		Remove: nil,
 		IsHash: true,
-	}
-	reqBodyBytes, err := json.Marshal(reqBody)
+	})
 	if err != nil {
-		return errors.AddContext(err, "failed to build request body")
+		return nil, nil, errors.AddContext(err, "failed to build request body")
 	}
 
+	// execute the request
 	url := fmt.Sprintf("http://%s:%d/skynet/blocklist?timeout=%s", api.staticNginxHost, api.staticNginxPort, skydTimeout)
-
-	api.staticLogger.Debugf("blockSkylinks: POST on %+s", url)
-	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(reqBodyBytes))
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(reqBody))
```

Comment:
Might warrant a generic `.post` method like we are already doing for get requests.

Reply:
I want to get rid of this entire file and move it to the client. I already have a bunch of that in my stash but I stashed it because of the size of the PR. That's probably the last PR I want to make on the blocker (after the ones that are open are merged). Seeing as I was planning on doing this already I'll do this in a F/U.

Reply:
fine with that?